packages/firebase_analytics/firebase_analytics/ios/Classes/FLTFirebaseAnalyticsPlugin.m

```diff
@@ -156,6 +160,27 @@ - (void)setDefaultEventParameters:(id)arguments
   result.success(nil);
 }
 
+- (void)initiateOnDeviceConversionMeasurementWithEmailAddressEmailAddress:(NSString *)emailAddress
+                                                           completion:
+                                                               (void (^)(FlutterError *_Nullable))
+                                                                   completion {
+  [FIRAnalytics initiateOnDeviceConversionMeasurementWithEmailAddress:emailAddress];
+
+  completion(nil);
+}
+
+- (void)setAutomaticDataCollectionEnabledAppName:(nonnull NSString *)appName
+                                         enabled:(nonnull NSNumber *)enabled
+                                      completion:
+                                          (nonnull void (^)(FlutterError *_Nullable))completion {
+  FIRApp *firebaseApp = [FLTFirebasePlugin firebaseAppNamed:appName];
+  if (firebaseApp) {
+    [firebaseApp setDataCollectionDefaultEnabled:[enabled boolValue]];
+  }
+
+  completion(nil);
+}
```

Comment:
I'm not seeing where this method is being called. You probably need to add something in the `handleMethodCall`

Reply:
@Lyokone I thought the same at first glance, but actually it's being called inside the `FirebaseAnalyticsHostApiSetup`

firebase_analytics_messages.g.m
```
void FirebaseAnalyticsHostApiSetup(id<FlutterBinaryMessenger> binaryMessenger,
                                   NSObject<FirebaseAnalyticsHostApi> *api) {
  {
    FlutterBasicMessageChannel *channel = [[FlutterBasicMessageChannel alloc]
           initWithName:
               @"dev.flutter.pigeon.firebase_analytics_platform_interface.FirebaseAnalyticsHostApi."
               @"initiateOnDeviceConversionMeasurementWithEmailAddress"
        binaryMessenger:binaryMessenger
                  codec:FirebaseAnalyticsHostApiGetCodec()];
    if (api) {
      NSCAssert([api respondsToSelector:@selector
                     (initiateOnDeviceConversionMeasurementWithEmailAddressEmailAddress:
                                                                             completion:)],
                @"FirebaseAnalyticsHostApi api (%@) doesn't respond to "
                @"@selector(initiateOnDeviceConversionMeasurementWithEmailAddressEmailAddress:"
                @"completion:)",
                api);
      [channel setMessageHandler:^(id _Nullable message, FlutterReply callback) {
        printf("Inside newMessageHandler");
        NSArray *args = message;
        NSString *arg_emailAddress = GetNullableObjectAtIndex(args, 0);
        [api initiateOnDeviceConversionMeasurementWithEmailAddressEmailAddress:arg_emailAddress
                                                                    completion:^(
                                                                        FlutterError
                                                                            *_Nullable error) {
                                                                      callback(
                                                                          wrapResult(nil, error));
                                                                    }];
      }];
    } else {
      [channel setMessageHandler:nil];
    }
  }
}
```

It receives the binary messenger and sets a parallel channel, which then calls to the appropriate function. If you run the example, you'll see that the function is properly executed and (if you add the print I wrote in this example) the print too.

In short, there's no need to call it in `handleMethodCall`, Pigeon already does it.

The ideal would be to have all the functions of firebase_analytics migrated into pigeon, then all of them would be called in the same place. But I think this would be overkill for this scenario and it would transform this non-breaking PR into a breaking one.